source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -343,6 +346,32 @@ std::string& TransformerInstance::random_for_pattern(const std::string& pattern)
   return found->second;
 }
 
+json TransformerInstance::raw_string_callback(const inja::Arguments &args) const {
+  // inja::Arguments is a std::vector<const json *>, so we can get the json
+  // value from the args directly.
+  const auto& input = args.at(0);
+
+  // make sure to bail if we're not working with a raw string value
+  if(!input->is_string()) {
+      return input->get_ref<const std::string&>();
+  }
+
+  auto val = input->dump();
+
+  // This block makes it such that a template must have surrounding " characters
+  // around the raw string. This is reasonable since we expect the value we get out of the
+  // context (body) to be placed in exactly as-is. HOWEVER, the behavior of the jinja
+  // filter is such that the quotes added by .dumps() are left in. For that reason,
+  // this callback is NOT named to_json to avoid confusion with that behavior.
+
+  // strip the leading and trailing " characters that are added by dump()
+  // if C++20 is adopted, val.starts_with and val.ends_with would clean this up a bit
+  val = val.substr(0,1) == "\"" && val.substr(val.length()-1,1) == "\""
+      ? val.substr(1, val.length()-2)
+      : val;
+  return val;
```

Comment:
did we see https://github.com/nlohmann/json/issues/1479#issuecomment-462900019

why is this preferred to their recommended solution? what about nested json that it will escape again?

Reply:
id like to see some tests with nested json; i suspect if we did that we would escape each child (we may not want) rather than just the outermost. but perhaps im wrong

Reply:
If the value was escaped as a string like so
```json
{"foo": "{\"bar\":\"baz\"}"}
```
then I would expect that the most atomic node that inja would parse would be to parse the "{\"bar\":\"baz\"}" as a string and _not_ then attempt to parse it as an object. That said, it's definitely a good question and worth testing

Reply:
I have added some tests with nested json to confirm the behavior. When valid JSON is fed in, regardless how much nesting in strings is done, there will not be double escaping 46f2a5f2597f208270d96a91568d61c2c17550a0

Reply:
thank you, fantastic